# Ejercicio 1

In [2]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_absolute_error, r2_score

import warnings

import funciones_auxiliares_machine_learning as aux

warnings.filterwarnings('ignore')
plt.rcParams['figure.figsize'] = (12, 6)
plt.style.use('seaborn')

df = pd.read_csv('ames_housing.csv').drop(columns='Unnamed: 0')

# Ejercicio 2

In [3]:
import pickle

modelo_propio = pickle.load(open('javier-rojas-actividad07.sav', 'rb'))
X_train = pickle.load(open('javier-rojas-X-train-actividad07.sav', 'rb'))
y_train = pickle.load(open('javier-rojas-y-train-actividad07.sav', 'rb'))
X_test = pickle.load(open('javier-rojas-X-test-actividad07.sav', 'rb'))
y_test = pickle.load(open('javier-rojas-y-test-actividad07.sav', 'rb'))

In [5]:
aux.reportar_metricas_regresion(modelo_propio, X_test, y_test)

|     |      valores |
|:----|-------------:|
| MAE | 24419.6      |
| R2  |     0.757795 |


## Reportar metricas de modelo de companero TODO

# Ejercicio 3

In [6]:
random_forest = RandomForestRegressor(random_state=1)
random_forest.fit(X_train, y_train)

RandomForestRegressor(random_state=1)

In [7]:
aux.reportar_metricas_regresion(random_forest, X_test, y_test)

|     |      valores |
|:----|-------------:|
| MAE | 19723.4      |
| R2  |     0.859388 |


- La tabla comparativa de resultados son las siguientes:


Modelo|MAE|R2
------|---|--
Grilla|24420|0.76|
RandomForest|19723|0.85|

- Por lo tanto, se puede observar un drastico aumento en el desempeno del modelo (disminucion del error, y aumento del poder explicativo r2).